apps/fipsinstall.c

```diff
@@ -415,6 +423,11 @@ int fipsinstall_main(int argc, char **argv)
         case OPT_DISALLOW_DRGB_TRUNC_DIGEST:
             fips_opts.drgb_no_trunc_dgst = 1;
             break;
+        case OPT_DSA_SIGN_CHECK:
+            if (!check_non_pedantic_fips(pedantic, "no_dsa_sign_check"))
```

Comment:
Hmmmm, this check isn't done consistently.  It's not present for EMS and DRBG digests e.g.  Of course it doesn't need to be done there because the option is turning it on which is compatible with pedantic.  However this leads to…

Doing it this way (using `no-` options), runs contrary to the OTC decision to make `fipsinstall` be compatible by default rather than pedantic.  As far as I'm aware that decision hasn't been revisited, so `fipsinstall` should produce a compatible installation _unless_ the `-pedantic` option is used.

You'll also nee to update `providers/fipsmodule.cnf.pl` which produces the installed `fipsmodule.cnf` file (when running `make install` et al).  This **should** produce a pedantic installation (based on a later OTC decision).

Reply:
I guess you mean util/mk-fipsmodule-cnf.pl
Why was this done like this - this seems to manually be doing what fipsinstall is doing??

Reply:
> I guess you mean util/mk-fipsmodule-cnf.pl Why was this done like this - this seems to manually be doing what fipsinstall is doing??

Because you cannot expect fipsinstall being present on cross-compile builds you cannot really run fipsinstall as part of make/make install.